### PR TITLE
Drop pre-3.0.0 deprecated APIs

### PR DESCRIPTION
**Pull request by seglo**
I dropped all deprecations. Let's discuss in the PR if any should be restored.

### CODE REVIEW

File: s3/src/main/resources/reference.conf [hunk 1]
```diff
@@ -74,19 +66,6 @@ alpakka.s3 {
     }
   }
```
Comment (seglo) [inline s3/src/main/resources/reference.conf:68]: I may have gone too far with dropping all the `path-style-access`/`access-path` code.

The only reason I can think is to support Minio users, but it was found that `endpoint-url` could be used to satisfy this case https://github.com/akka/alpakka/issues/2364.

**Review by ennru — APPROVED** [review]
Nice cleanup.
LGTM.

**Comment by ennru** [comment]
S3 tests are unhappy.

File: s3/src/main/resources/reference.conf (hunk 1 above)
Comment (seglo) [reply s3/src/main/resources/reference.conf:68]: Let's keep these deprecated APIs for now. AWS has decided to support path-style-access indefinitely, so until it's dropped, and there are alternatives for issues like #2620, we should have the option available for users.

Comment (seglo) [reply s3/src/main/resources/reference.conf:68]: > Update (September 23, 2020) – Over the last year, we’ve heard feedback from many customers who have asked us to extend the deprecation date. Based on this feedback we have decided to delay the deprecation of path-style URLs to ensure that customers have the time that they need to transition to virtual hosted-style URLs.
>
> We have also heard feedback from customers that virtual hosted-style URLs should support buckets that have dots in their names for compatibility reasons, so we’re working on developing that support. Once we do, we will provide at least one full year prior to deprecating support for path-style URLs for new buckets.

https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/

**Pull request merged**